src/core/classes/storage.ts

```diff
@@ -0,0 +1,36 @@
+import { JsonObject } from "..";
+import { FileOrBuffer } from "../..";
+import { IStorage } from "../interfaces/IStorage";
+
+export class Storage {
+  private storage: IStorage;
+
+  constructor(storage: IStorage) {
+    this.storage = storage;
+  }
+
+  public async fetch(hash: string): Promise<Record<string, any>> {
+    return this.storage.get(hash);
+  }
+
+  public async upload(data: FileOrBuffer[] | JsonObject[]): Promise<string> {
```

Comment:
hmmm can this be `(FileOrBuffer | JsonObject)[] | (FileOrBuffer | JsonObject)`

so i can pass a single thing if i want to?

also the return type should give me both the base uri and the individual uris i think right?